tmt/steps/prepare/artifact/providers/__init__.py

```diff
@@ -195,6 +199,117 @@ def _filter_artifacts(self, exclude_patterns: list[Pattern[str]]) -> Iterator[Ar
                 yield artifact
 
 
+@container
+class Repository:
+    """
+    Thin wrapper/holder for .repo file content
+    """
+
+    #: Content of the repository
+    content: str
+    #: Uniquely identifiable name
+    name: str
+    #: repository_ids present in the .repo file
+    repo_ids: list[str] = simple_field(default_factory=list[str])
+
+    def __post_init__(self) -> None:
+        """
+        Extract repository IDs from the .repo file content after initialization.
+
+        :raises GeneralError: If the content is malformed or no repository
+            sections are found.
+        """
+        config = configparser.ConfigParser()
+        try:
+            config.read_string(self.content)
+            sections = config.sections()
+            if not sections:
+                raise GeneralError(
+                    f"No repository sections found in the content for '{self.name}'."
+                )
+            # Store the parsed sections in our private attribute
+            self.repo_ids = sections
+        except configparser.MissingSectionHeaderError:
+            raise GeneralError(f"No repository sections found in the content for '{self.name}'.")
+        except configparser.Error as error:
+            raise GeneralError(
+                f"Failed to parse the content of repository '{self.name}'. "
+                "The .repo file may be malformed."
+            ) from error
+
+    @classmethod
+    def from_url(cls, url: str, name: Optional[str] = None) -> "Repository":
+        """
+        Create a Repository instance by fetching content from a URL.
+
+        :param url: The URL to fetch the repository content from.
+        :param name: Optional name for the repository. If not provided,
+            derived from the URL.
+        :returns: A Repository instance.
+        :raises GeneralError: If fetching or parsing fails.
+        """
+        try:
+            with tmt.utils.retry_session() as session:
```

Comment:
```suggestion
            with tmt.utils.retry_session(logger=logger) as session:
```
And add it to the required variables in `from_url`